src/internal/storage/chunk/reader.go

```diff
@@ -7,29 +7,49 @@ import (
 	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
 	"github.com/pachyderm/pachyderm/v2/src/internal/errutil"
 	"github.com/pachyderm/pachyderm/v2/src/internal/storage/kv"
+	"github.com/sirupsen/logrus"
 )
 
 // Reader reads data from chunk storage.
 type Reader struct {
-	ctx      context.Context
-	client   Client
-	memCache kv.GetPut
-	dataRefs []*DataRef
+	ctx         context.Context
+	client      Client
+	memCache    kv.GetPut
+	dataRefs    []*DataRef
+	offsetBytes int64
 }
 
-func newReader(ctx context.Context, client Client, memCache kv.GetPut, dataRefs []*DataRef) *Reader {
-	return &Reader{
+type ReaderOption func(*Reader)
+
+func WithOffsetBytes(offsetBytes int64) ReaderOption {
+	return func(r *Reader) {
+		r.offsetBytes = offsetBytes
+	}
+}
+
+func newReader(ctx context.Context, client Client, memCache kv.GetPut, dataRefs []*DataRef, opts ...ReaderOption) *Reader {
+	r := &Reader{
 		ctx:      ctx,
 		client:   client,
 		memCache: memCache,
 		dataRefs: dataRefs,
 	}
+	for _, opt := range opts {
+		opt(r)
+	}
+	return r
 }
 
 // Iterate iterates over the data readers for the data references.
 func (r *Reader) Iterate(cb func(*DataReader) error) error {
+	offset := r.offsetBytes
 	for _, dataRef := range r.dataRefs {
-		dr := newDataReader(r.ctx, r.client, r.memCache, dataRef)
+		if dataRef.SizeBytes <= offset {
+			offset -= dataRef.SizeBytes
+			continue
+		}
+		dr := newDataReader(r.ctx, r.client, r.memCache, dataRef, offset)
+		offset = 0
```

Comment:
Is this assignment necessary?

Reply:
yep! if it's omitted and there are more dataRefs to iterate through, we will apply that small offset to them too